lib/modules/manager/bazel-module/fragments.ts

```diff
@@ -225,45 +235,52 @@ export type ValueFragment =
 export type ChildFragments = Record<string, ValueFragment>;
 export type Fragment = ValueFragment | AttributeFragment;
 
-// Fragments Class
+// Static Functions
 
-export class Fragments {
-  static typeError(expected: string, actual: string): Error {
-    return new Error(`Expected type ${expected}, but was ${actual}.`);
+function as<I extends ValueFragmentInterface, F>(
+  data: unknown,
+  fragCtor: new (...args: any[]) => F,
+  iTypeGuard: (v: ValueFragmentInterface) => v is I,
+  fromFn: (frag: I) => F
+): F {
+  if (data instanceof fragCtor) {
+    return data;
   }
+  const fragInterface = ValueFragmentSchema.parse(data);
+  if (iTypeGuard(fragInterface)) {
+    return fromFn(fragInterface);
+  }
+  throw new Error(`Incorrect value fragment type: ${fragInterface.type}.`);
```

Comment:
```suggestion
  throw new Error(`Wrong value fragment type: ${fragInterface.type}.`);
```

The Plain Language Styleguide (that we adopt) says to: _use simple words and phrases_. Replace `Incorrect` with `Wrong`. 😉 

I don't understand this error message. Is the error is about the _value_, the _type_, or the "value fragement's type"?

Reply:
I added some additional context to the error message. Let me know if that works for you.